libraries/chain/controller.cpp

```diff
@@ -2917,7 +2931,8 @@ struct controller_impl {
          }
          
          auto do_push = [&](auto& fork_db, auto& head) {
-            fork_db.add( bsp );
+            if constexpr (std::is_same_v<BSP, typename std::decay_t<decltype(head)>>)
```

Comment:
That is really ugly, would be nice to find a way not to have these `if constexpr`.